packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -708,12 +763,343 @@ export const prepareVaultManagerKit = (
         getPublicFacet() {
           return this.facets.collateral;
         },
+
+        lockOraclePrices() {
+          const { state, facets } = this;
+          const { self } = facets;
+
+          // XXX: 'twould be better if collateralQuote were updated lazily when it
+          // changes by more than X%, and we use a cached value.  see #6946
+          return E.when(self.getCollateralQuote(), quote => {
+            trace(`lockPrice`, getAmountIn(quote), getAmountOut(quote));
+            // @ts-expect-error declared PriceQuote | undefined. TS thinks undefined.
+            state.lockedQuote = quote;
+            return quote;
+          });
+        },
+        async liquidateVaults(auctionPF) {
+          const { state, facets } = this;
+          const { lockedQuote, compoundedInterest } = state;
+          const oraclePriceAtStartP = facets.self.getCollateralQuote();
+
+          assert(factoryPowers && prioritizedVaults && zcf);
+          lockedQuote ||
+            Fail`Must have locked a quote before liquidating vaults.`;
+          assert(lockedQuote); // redundant with previous line
+
+          const crKey = normalizedCrKey(lockedQuote, compoundedInterest);
+
+          trace(
+            `Liquidating vaults worse than`,
+            normalizedCr(lockedQuote, compoundedInterest),
+          );
+          const {
+            totalDebt = zeroDebt,
+            totalCollateral = zeroCollateral,
+            vaultData,
+            liqSeat,
+          } = getLiquidatableVaults(
+            zcf,
+            crKey,
+            prioritizedVaults,
+            liquidatingVaults,
+          );
+          if (vaultData.getSize() === 0) {
+            return;
+          }
+          trace(
+            ' Found vaults to liquidate',
+            liquidatingVaults.getSize(),
+            totalCollateral,
+          );
+
+          facets.helper.liquidatingUpdate(totalDebt, totalCollateral);
+          facets.helper.updateMetrics();
+
+          const { userSeatPromise, deposited } = await E.when(
+            E(auctionPF).getDepositInvitation(),
+            depositInvitation =>
+              offerTo(
+                zcf,
+                depositInvitation,
+                harden({ Minted: 'Currency' }),
+                harden({ give: { Collateral: totalCollateral } }),
+                liqSeat,
+              ),
+          );
+
+          // This can be a long wait.
```

Comment:
how long in a) common case and b) extreme case?

Reply:
nominal is 40 minutes, though the EC can choose different parameters. Whatever nominal is, if it takes much longer, something is broken. But when the chain comes back, everything should reconnect.

Reply:
ok please document that, what the parameter is which defines it